packages/plugin-e2e/src/models/Panel.ts

```diff
@@ -10,6 +10,26 @@ export class Panel extends GrafanaPage {
     super(ctx);
   }
 
+  /**
+   * Returns a locator that resolves element(s) that contain the field name(s) that are currently displayed in the panel.
+   *
+   * Can be used to assert the field names displayed in the panel visualization. e.g
+   * await expect(panelEditPage.panel.getFieldNames()).toHaveValues(['Month', 'Stockholm', 'Berlin', 'Log Angeles']);
+   */
+  getFieldNames(): Locator {
+    return this.locator().locator('[role="columnheader"]');
+  }
+
+  /**
+   * Returns a locator that resolves element(s) that contain the value(s) that are currently displayed in the panel.
+   *
+   * Can be used to assert the values displayed in the panel visualization. e.g
+   * await expect(panelEditPage.panel.getData()).toContainText(['1', '4', '14']);
+   */
+  getData(): Locator {
+    return this.locator().locator('[role="cell"]');
+  }
+
```

Comment:
FYI I'm working on adding proper e2e selectors for the table panel in grafana core. 

Even though people will probably want to use the Table panel in most cases, it won't cover the scenario when you want to assert on panel (that is not table panel) data in a dashboard, as you can't toggle to `Table view` from there.  Not high priority, but we could easily add parsing functionality to some of the most common panel types. E.g

```ts
getData(): Locator {
    switch (this.visualizationType) {
      case 'table':
        return this.locator().locator('[role="cell"]');
      case 'logs':
        return this.locator().locator(selectors.components.Panels.Panel.Logs.logRow);
      case 'gauge':
        return this.locator().locator(selectors.components.Panels.Panel.Gauge.value);
      default:
        throw new Error(`No data parser implemented for this visualization type: ${this.visualizationType}`);
    }
  }
```